homeassistant/components/synology_dsm/entity.py

```diff
@@ -85,35 +90,45 @@ def __init__(
         self._device_model: str | None = None
         self._device_firmware: str | None = None
         self._device_type = None
+        storage = api.storage
+        information = api.information
+        network = api.network
+        assert information is not None
+        assert storage is not None
+        assert network is not None
 
         if "volume" in description.key:
-            volume = self._api.storage.get_volume(self._device_id)
+            assert self._device_id is not None
+            volume = storage.get_volume(self._device_id)
+            assert volume is not None
             # Volume does not have a name
             self._device_name = volume["id"].replace("_", " ").capitalize()
             self._device_manufacturer = "Synology"
-            self._device_model = self._api.information.model
-            self._device_firmware = self._api.information.version_string
+            self._device_model = information.model
+            self._device_firmware = information.version_string
             self._device_type = (
                 volume["device_type"]
                 .replace("_", " ")
                 .replace("raid", "RAID")
                 .replace("shr", "SHR")
             )
         elif "disk" in description.key:
-            disk = self._api.storage.get_disk(self._device_id)
+            assert self._device_id is not None
+            disk = storage.get_disk(self._device_id)
+            assert disk is not None
             self._device_name = disk["name"]
             self._device_manufacturer = disk["vendor"]
             self._device_model = disk["model"].strip()
-            self._device_firmware = disk["firm"]
-            self._device_type = disk["diskType"]
+            self._device_firmware = disk["firm"]  # type: ignore[typeddict-item]
+            self._device_type = disk["diskType"]  # type: ignore[typeddict-item]
```

Comment:
@mib1185 2.4.1 fixes the other but it looks like these are still missing

Reply:
added these, too and release 2.4.2